tuxemon/event/conditions/check_evolution.py

```diff
@@ -41,60 +41,75 @@ def test(self, session: Session, condition: MapCondition) -> bool:
-                    # use the item on the monster
-                    if evolution.item:
-                        _evolution = False
-                    if _evolution:
-                        evolving.append((monster, evolved))
-
-        if evolving:
-            character.pending_evolutions = evolving
-        return len(evolving) > 0
+                    if self.can_evolve(monster, evolution):
+                        evolved_monster = Monster()
+                        evolved_monster.load_from_db(evolution.monster_slug)
+                        evolving_monsters.append((monster, evolved_monster))
+
+        if evolving_monsters:
+            character.pending_evolutions = evolving_monsters
+
+        return len(evolving_monsters) > 0
+
+    def can_evolve(
+        self, monster: Monster, evo: MonsterEvolutionItemModel
+    ) -> bool:
+        # Check if the evolution is actually possible
+        if evo.monster_slug == monster.slug:
+            return False
+
+        conditions = [
+            evo.at_level
+            and compare("greater_or_equal", monster.level, evo.at_level)
+            or False,
+            evo.gender and evo.gender == monster.gender or False,
+            evo.element and monster.has_type(evo.element) or False,
+            evo.tech and self.character.has_tech(evo.tech) or False,
+            evo.inside and evo.inside == self.client.map_inside or False,
+            evo.traded and evo.traded == monster.traded or False,
+        ]
+
+        # Check if the monster's stats meet the evolution conditions
+        if evo.stats:
+            params = evo.stats.split(":")
+            operator = params[1]
+            stat1 = monster.return_stat(StatType(params[0]))
+            stat2 = monster.return_stat(StatType(params[2]))
+            conditions.append(compare(operator, stat1, stat2))
+
+        # Check if the monster's game variables meet the evolution conditions
+        if evo.variable:
+            parts = evo.variable.split(":")
+            key, value = parts[:2]
+            conditions.append(
+                key in self.character.game_variables
+                and self.character.game_variables[key] == value
+            )
+
+        # Check if the monster has taken the required number of steps
+        if evo.steps:
+            result = evo.steps - int(monster.steps)
+            conditions.append(result == 0)
+            monster.steps += 1
+            monster.levelling_up = True
+            monster.got_experience = True
+
+        # Check if the monster's bond meets the evolution conditions
+        if evo.bond:
+            parts = evo.bond.split(":")
+            operator, value = parts[:2]
+            _bond = monster.bond
+            conditions.append(compare(operator, _bond, int(value)))
+
+        # If the evolution requires an item, do not evolve
+        if evo.item:
+            return False
+        # The monster can evolve if all conditions are met
```

Comment:
"all conditions are met".

"all" in comment contradicts `any(conditions)`

Reply:
ok